### PR TITLE
#2948 - Make the bootstrapped S3 bucket compliant with SecurityHub rules. (S3…

### PR DESCRIPTION
**Pull request by ackjewtn**
….4, S3.5)

#### Which issue(s) does this change fix?

#2948 

#### Why is this change necessary?
To be compliant with https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html

#### How does it address the issue?
Adds the appropriate properties to the S3 bucket in the CloudFormation template, hence creating a compliant S3 bucket.

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/lib/bootstrap/bootstrap.py
Comment (hawflau) [inline samcli/lib/bootstrap/bootstrap.py:89]: According to [here](https://aws.amazon.com/premiumsupport/knowledge-center/s3-bucket-policy-for-config-rule/), do we also need to add the bucket arn in addition to all objects under the bucket?
For example:
```
      "Resource": [
        "arn:aws:s3:::DOC-EXAMPLE-BUCKET",
        "arn:aws:s3:::DOC-EXAMPLE-BUCKET/*"
      ]
```

Comment (ackjewtn) [reply samcli/lib/bootstrap/bootstrap.py]: I agree, my bad. Fixed in https://github.com/aws/aws-sam-cli/pull/2953/commits/b241585acfe391f06c581290464da1888063255c

Comment (hawflau) [reply samcli/lib/bootstrap/bootstrap.py]: Thanks!

**Review by hawflau — APPROVED** [review]
LGTM!

We probably can have another PR to add integration test to verify the managed stack and the S3 bucket.

**Comment by dza89** [comment]
Can this be merged? Waiting for this as well.
Thanks!

**Comment by ackjewtn** [comment]
@hawflau Are you guys waiting on anything from me to get his merged in?

**Comment by jfuss** [comment]
@ackjewtn Thanks for pinging on this and contributing. We really appreciate it.

I don't think there is anything specific (at least from what I can tell from feedback provided). What likely happened is this slipped through the cracks and didn't get a second review (we require 2 approvals).

I am going to assign this to myself to remind me to ask the team on Monday. Will respond back here once I have an understanding if my guess is true.

We may want to consider how to update existing buckets we created but that is out of scope for this (imo). 

Handling the conflicts, would be awesome if you have some time but I can do that too if you don't have time.

**Comment by ackjewtn** [comment]
@jfuss No problem. I've fixed the conflict and applied the fixes to the template again. Seems like all the tests except one in `make-pr`  has gone through and based on the build history, this specific one has a tendency to get stuck, time out and fail, so i'm assuming that all is good.

Let me know if anything else is needed from me.

**Comment by jfuss** [comment]
@ackjewtn Thanks for updating the PR. I re-ran the stuck job and everything is good now.

**Review by SimonCMoore — APPROVED** [review]

**Pull request merged**
